### ISSUE TITLE
NVDA becomes silent when reinstalling an add-on while the "Add-ons manager" dialog is open

### ISSUE THREAD

**Issue by paulber19** [nvaccess/nvda#14551]
### Actual behavior:

- install an add-on from its installation file
- restart NVDA
- open the add-ons manager
- launch the installation of the add-on again from its installation file
- answer yes to the questions, including the one asking to restart NVDA.
- NVDA does not restart and goes silent.

By launching the narrator, we realize that the NVDA window which asks to restart is always displayed and that it is enough to press one of the buttons for NVDA to restart.

This does not happen if the add-on  is not already installed.
Test with NVDA 2022.4 and 2021.1
Error in the log:
ERROR - core.restart (19:03:44.921) - MainThread (3840):
NVDA already in process of exiting, this indicates a logic error.

### Expected behavior:

NVDA should behave as when the add-on was first installed. NVDA must reboot.
#### NVDA installed/portable/running from source:
NVDA installed and portable
#### NVDA version:
NVDA 2022.4 and NVDA 2021.1
#### Windows version:
W10 22h2
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

**Comment by seanbudd** [comment]
Closing as duplicate of #14525
